File: patterns/strategy.py
class AbsInterpolator(object):
	def interpolate(x):
		raise NotImplementedError('interpolate method has not been implemented')

class LinearInterpolator(AbsInterpolator):
	def __init__(self,xvec,yvec):
		pass
	def interpolate(self,x):
		pass

class SplineInterpolator(AbsInterpolator):
	def __init__(self,xvec,yvec):
		pass
	def interpolate(self,x):
		pass

File: patterns/strategy.cc
#include <vector>

class AbsInterpolator {
public:
	virtual double interpolate(double x) const = 0;
};

class LinearInterpolator : public AbsInterpolator {
	LinearInterpolator(std::vector<double> const &xvec, std::std::vector<double> const &yvec);
	double interpolate(double x) const;
};

class SplineInterpolator : public AbsInterpolator {
	SplineInterpolator(std::vector<double> const &xvec, std::std::vector<double> const &yvec);
	double interpolate(double x) const;
};
